src/main/java/io/confluent/examples/streams/JsonToAvroExample.java

```diff
@@ -0,0 +1,133 @@
+ * The already running example application (step 3) will automatically process this input data and
+ * write the results to the output topic. The {@link JsonToAvroExampleDriver} will print the
+ * results from the output topic
+ * <pre>
+ * {@code
+ * # Here: Write input data using the example driver.  Once the driver has stopped generating data,
+ * # you can terminate it via Ctrl-C.
+ * $ java -cp target/kafka-streams-examples-5.2.0-SNAPSHOT-standalone.jar io.confluent.examples.streams.JsonToAvroExampleDriver
+ * }
+ * </pre>
+ */
+public class JsonToAvroExample {
+
+  static final String JSON_SOURCE_TOPIC = "json-source";
+  static final String AVRO_SINK_TOPIC = "avro-sink";
+
+  public static void main(final String[] args) {
+    final String bootstrapServers = args.length > 0 ? args[0] : "localhost:9092";
+    final String schemaRegistryUrl = args.length > 1 ? args[1] : "http://localhost:8081";
+    final KafkaStreams streams = buildJsonToAvroStream(
+        bootstrapServers,
+        schemaRegistryUrl
+    );
+    streams.start();
+
+    // Add shutdown hook to respond to SIGTERM and gracefully close Kafka Streams
+    Runtime.getRuntime().addShutdownHook(new Thread(streams::close));
+  }
+
+  static KafkaStreams buildJsonToAvroStream(final String bootstrapServers,
+      final String schemaRegistryUrl) {
+    final Properties streamsConfiguration = new Properties();
+
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "json-to-avro-stream-conversion");
+    streamsConfiguration.put(StreamsConfig.CLIENT_ID_CONFIG, "json-to-avro-stream-conversion-client");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    // Where to find the Confluent schema registry instance(s)
+    streamsConfiguration.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl);
+
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
+    streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 10 * 1000);
+
+    final ObjectMapper objectMapper = new ObjectMapper();
+    final TypeReference<HashMap<String, String>> typeReference = new TypeReference<HashMap<String, String>>() {};
+
+    final StreamsBuilder builder = new StreamsBuilder();
+
+    // read the source stream
+    final KStream<String, String> jsonToAvroStream = builder.stream(JSON_SOURCE_TOPIC, Consumed.with(Serdes.String(), Serdes.String()));
+    jsonToAvroStream.mapValues(v -> {
+      WikiFeed wikiFeed = null;
+      try {
+        final Map<String, String> jsonMap = objectMapper.readValue(v, typeReference);
+        wikiFeed = new WikiFeed(jsonMap.get("user"), Boolean.valueOf(jsonMap.get("is_new")), jsonMap.get("content"));
```

Comment:
You don't need to do the typereference thing to just get a map. It would be a better (and more general, and more efficient, too) experience to just use `readTree` and then query the json data structure like `json.get("is_new").asBoolean()`, etc.

Reply:
ack